src/bindings/python/tests/runtime.py

```diff
@@ -110,28 +108,24 @@ def __call__(self, *input_values: NumericData) -> List[NumericData]:
         # Input validation
         if len(input_values) < len(self.parameters):
             raise UserInputError(
-                "Expected %s params, received not enough %s values.", len(self.parameters), len(input_values),
+                "Expected %s params, received not enough %s values.",
+                len(self.parameters),
+                len(input_values),
             )
 
         param_names = [param.friendly_name for param in self.parameters]
         input_shapes = [get_shape(input_value) for input_value in input_values]
-
         if self.network_cache.get(str(input_shapes)) is None:
             function = self.function
-            if self.function.is_dynamic():
-                function = function.clone()
-                function.reshape(dict(zip(param_names, [PartialShape(i) for i in input_shapes])))
             self.network_cache[str(input_shapes)] = function
```

Comment:
I am not sure if that is correct, since later in else block we are recovering function as `function = self.network_cache[str(input_shapes)]` so depending if we end up in `if` block or `else` block, `function` variable will be either present or not